modules/core/04-channel/keeper/timeout_test.go

```diff
@@ -115,6 +162,25 @@ func (suite *KeeperTestSuite) TestTimeoutPacket() {
 			err = path.EndpointA.SetChannelState(types.CLOSED)
 			suite.Require().NoError(err)
 		}, false},
+		{"channel in TRYUPGRADE and FLUSH_COMPLETE", func() {
+			expError = types.ErrInvalidChannelState
+			suite.coordinator.Setup(path)
+
+			timeoutHeight := clienttypes.GetSelfHeight(suite.chainB.GetContext())
+
+			sequence, err := path.EndpointA.SendPacket(timeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+			suite.Require().NoError(err)
+			packet = types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, timeoutHeight, disabledTimeoutTimestamp)
+			// need to update chainA's client representing chainB to prove missing ack
+			err = path.EndpointA.UpdateClient()
+			suite.Require().NoError(err)
+
+			// Move channel to TRYUPGRADE and to FLUSH_COMPLETE manually.
```

Comment:
due to the positioning of the check in `TimeoutPacket` (after we check if `len(commitment) == 0` it doesn't seem feasable we can check the combination `[TRYUPGRADE/FLUSHCOMPLETE]` w/o setting them manually.